ocean_lib/assets/trusted_algorithms.py

```diff
@@ -63,141 +82,110 @@ def create_publisher_trusted_algorithms(
 
 @enforce_types
 def add_publisher_trusted_algorithm(
-    asset_or_did: Union[str, V3Asset], algo_did: str, metadata_cache_uri: str
+    asset_or_did: Union[str, V3Asset, V4Asset], algo_did: str, metadata_cache_uri: str
 ) -> list:
     """
     :return: List of trusted algos
     """
-    if isinstance(asset_or_did, V3Asset):
+    if isinstance(asset_or_did, (V3Asset, V4Asset)):
         asset = asset_or_did
     else:
         asset = resolve_asset(asset_or_did, metadata_cache_uri=metadata_cache_uri)
 
-    compute_service = asset.get_service(ServiceTypes.CLOUD_COMPUTE)
+    compute_service = asset.get_service("compute")
     assert (
         compute_service
     ), "Cannot add trusted algorithm to this asset because it has no compute service."
-    privacy_values = compute_service.attributes["main"].get("privacy")
-    if not privacy_values:
-        privacy_values = {}
-        compute_service.attributes["main"]["privacy"] = privacy_values
 
-    assert isinstance(privacy_values, dict), "Privacy key is not a dictionary."
-    trusted_algos = privacy_values.get("publisherTrustedAlgorithms", [])
-    # remove algo_did if already in the list
-    trusted_algos = [ta for ta in trusted_algos if ta["did"] != algo_did]
-
-    # now add this algo_did as trusted algo
     algo_ddo = resolve_asset(algo_did, metadata_cache_uri=metadata_cache_uri)
-    trusted_algos.append(generate_trusted_algo_dict(asset_or_did=algo_ddo))
+    generated_trusted_algo_dict = generate_trusted_algo_dict(algo_ddo)
 
-    # update with the new list
-    privacy_values["publisherTrustedAlgorithms"] = trusted_algos
-    assert (
-        compute_service.attributes["main"]["privacy"] == privacy_values
-    ), "New trusted algorithm was not added. Failed when updating the privacy key. "
-    return trusted_algos
+    if asset.version.startswith("4."):
```

Comment:
the content of these blocks is identical. The point was to remove the "if/else" and you can now do that.